x-pack/docs/en/security/authentication/configuring-pki-realm.asciidoc

```diff
@@ -179,3 +218,78 @@ NOTE: The PKI realm supports
 alternative to role mapping.
 
 --
+
+[float]
+[role="xpack"]
+[[pki-realm-for-proxied-clients]]
+==== PKI authentication for clients connecting to {kib}
+
+Clients must normally connect directly to {es} because the {es} PKI realm
+relies on the node's network interface to do the SSL/TLS authentication and to
+also internally forward the client's certificate chain for the realm to be able
+to optionally validate it. If SSL/TLS authenticatication is to be performed
+by {kib}, the PKI realm must be configured to permit delegation.
+
+Specifically, when clients presenting X.509 certificates connect to {kib},
+{kib} performs the SSL/TLS authentication. {kib} then forwards the client's
+certificate chain, by calling an {es} API, to have them further validated by
+the PKI realms that have been configured for delegation.
+
+To permit authentication delegation for a specific {es} PKI realm, start by
+configuring the realm for the usual case, as detailed in the
+<<pki-realm-for-direct-clients>>
+section. However, you must make sure that you have explicitly configured
+a `truststore` (or, equivalently `certificate_authorities`) even though it is
+the same trust configuration that you have configured on the network layer.
+Afterwards, simply toggle the `delegation.enabled` realm setting to `true`.
+This realm is now allowed to validate delegated PKI authentication (after
+restarting {es}).
+
+NOTE: PKI authentication delegation requires that
+the `xpack.security.authc.token.enabled` setting is `true`.
+
+NOTE: {kib} also needs to be configured to allow PKI certificate authentication.
+
+A PKI realm with `delegation.enabled` still works unchanged for clients
+connecting directly to {es}. Directly authenticated users, and users that are PKI
+authenticated by delegation to {kib} both follow the same
+{stack-ov}/mapping-roles.html[role mapping rules] or
+{stack-ov}/realm-chains.html#authorization_realms[authorization realms
+configurations].
+
+However, if you use the <<security-role-mapping-apis,role mapping APIs>>,
+you can distinguish between users that are authenticated by delegation and
+users that are authenticated directly. The former have the
+extra fields `pki_delegated_by_user` and `pki_delegated_by_realm` in the user's
+metadata. In the common setup, where authentication is delegated to {kib}, the
+values of these fields are `kibana` and `reserved`, respectively. For example,
+the following role mapping rule will assign the `role_for_pki1_direct` role to
+all users that have been authenticated directly by the `pki1` realm, by
+connecting to {es} instead of going through {kib}:
+
+[source,js]
+--------------------------------------------------
+PUT /_security/role_mapping/direct_pki_only
+{
+  "roles" : [ "role_for_pki1_direct" ],
+  "rules" : {
+    "all": [
+      {
+        "field": {"realm.name": "pki1"}
+      },
+      {
+        "except": {
+          "field": {
+            "metadata.pki_delegated_by_user": null <1>
+          }
+        }
```

Comment:
Isn't this example backwards (or am I confused?)
This is saying:
_The user has the "role_for_pki1_direct" role if they are authenticated by the "pki1" realm, **unless** they have a null delegation_
That _unless_ should be an _and_ shouldn't it?

Reply:
Oops, you're right! This is what I've changed it to:
```
{
  "roles" : [ "role_for_pki1_direct" ],
  "rules" : {
    "all": [
      {
        "field": {"realm.name": "pki1"}
      },
      {
        "field": {
          "metadata.pki_delegated_by_user": null <1>
        }
      }
    ]
  },
  "enabled": true
}
```